src/handlers.rs

```diff
@@ -701,3 +702,20 @@ fn handle_symlink(i: *const isolate, base: &msg::Base) -> Box<Op> {
     }()))
   }
 }
+
+fn handle_truncate(i: *const isolate, base: &msg::Base) -> Box<Op> {
+  let deno = from_c(i);
+  if !deno.flags.allow_write {
+    return odd_future(permission_denied());
+  }
+  let msg = base.msg_as_truncate().unwrap();
+  let name = msg.name().unwrap();
+  let len = msg.len();
+  Box::new(futures::future::result(|| -> OpResult {
+    debug!("handle_truncate {} {}", name, len);
+    let mut options = fs::OpenOptions::new();
+    let f = options.write(true).open(name)?;
+    f.set_len(len as u64)?;
```

Comment:
This is how you truncate in rust? This seems odd. 
Can you link me to documentation? I can't find anything on this....

Reply:
I believe he meant this call: [`File.set_len`](https://doc.rust-lang.org/std/fs/struct.File.html#method.set_len)

Reply:
Thanks

Reply:
@ry  Here is link. [https://doc.rust-lang.org/std/fs/struct.File.html#method.set_len](https://doc.rust-lang.org/std/fs/struct.File.html#method.set_len)     In order for use `set_len`, write access must be used.